src/custom-errror.ts

```diff
@@ -8,3 +11,21 @@ export class CustomError extends Error {
     this.details = details;
   }
 }
+
+export function formatError(formattedError: GraphQLFormattedError, error: unknown) {
+  let originalError = error;
+
+  if (error instanceof GraphQLError) {
+    originalError = unwrapResolverError(error);
+  }
+
+  if (originalError instanceof CustomError) {
+    return {
+      message: originalError.message,
+      code: originalError.code,
+      details: originalError.details,
+    };
+  } else {
+    return { message: 'Erro interno', code: 499, details: (error as Error).message };
+  }
```

Comment:
esse ultimo else pode ser removido já que o if é um retorno, né? e no caso de erro interno é comum utilizar o código 500
```suggestion
  }
  
  return { message: 'Erro interno', code: 500, details: (error as Error).message };
```